src/utility/math.ts

```diff
@@ -21,28 +21,26 @@ export function lerp(a: number, b: number, t: number): number {
 
 /**
  * Computes whether two objects are on a crash course and
- * will collide in the near future
+ * in how many seconds will they collide.
  * @param colliderA Collider for first object
  * @param colliderB Collider for second object
  * @param forwardA Forward vector of first object
  * @param forwardB Forward vector of second object
- * @param imminencyThreshold Threshold, in seconds, determining how soon should crash happen to be considered imminent (when function returns true)
- * @returns By default, function returns true if two objects
- * are on a crash course and will collide within the next second.
+ * @returns Returns time in seconds when collision will occur or
+ * undefined if no collision can occur.
  */
-export function isCrashImminent(
+export function getTimeBeforeCollision(
     colliderA: CircleCollider, 
     colliderB: CircleCollider, 
     forwardA: Vector, 
     forwardB: Vector, 
-    imminencyThreshold = 1
-) {
+) : number | undefined {
```

Comment:
Better return `null` instead of `undefined`. `undefined` is tricky and there are 2 types of runtime undefined (nonexistent property vs. existing property with the value of undefined (really messy stuff)). User code should *very rarely* return undefined.